src/componentsNext/Filepicker/index.tsx

```diff
@@ -0,0 +1,146 @@
+import { Breadcrumb, BreadcrumbRoot } from "~/design/Breadcrumbs";
+import {
+  Filepicker,
+  FilepickerClose,
+  FilepickerHeading,
+  FilepickerList,
+  FilepickerListItem,
+  FilepickerSeparator,
+} from "~/design/Filepicker";
+import { FolderUp, Home } from "lucide-react";
+import { Fragment, useState } from "react";
+
+import Button from "~/design/Button";
+import { ButtonBar } from "~/design/ButtonBar";
+import Input from "~/design/Input";
+import { analyzePath } from "~/util/router/utils";
+import { fileTypeToIcon } from "~/api/tree/utils";
+import { useNodeContent } from "~/api/tree/query/node";
+import { useTranslation } from "react-i18next";
+
+const FilepickerMenu = ({
+  namespace,
+  defaultPath,
+  onChange,
+}: {
+  namespace: string;
+  defaultPath?: string;
+  onChange: (newValue: string) => void;
+}) => {
+  const [path, setPath] = useState(defaultPath ? defaultPath : "/");
+
+  const givenNamespace = namespace ? namespace : "";
+
+  const { data, isFetched, isError } = useNodeContent({
+    path,
+    givenNamespace,
```

Comment:
The `givenNamespace` sounds a little bit strange when reading it here. I would expect it so be just `namespace`. Inside the hook, it makes perfect sense, since there is a variable naming collision. But what we could do is the following:
- calling the property of the parameter object `namespace`
- when we destructure it inside the hook, we rename it to `givenNamespace`
```
  ...
  enabled = true,
  namespace: givenNamespace,
  ...
```